internal/app/init/main.go

```diff
@@ -287,54 +307,60 @@ func startKubernetesServices(data *userdata.UserData) {
 	)
 }
 
+func reboot() {
+	// See http://man7.org/linux/man-pages/man2/reboot.2.html.
+	unix.Sync()
+	// nolint: errcheck
+	unix.Reboot(rebootFlag)
+
+	if *inContainer {
+		return
+	}
+
+	select {}
+}
+
 func recovery() {
 	if r := recover(); r != nil {
 		log.Printf("recovered from: %+v\n", r)
 		for i := 10; i >= 0; i-- {
 			log.Printf("rebooting in %d seconds\n", i)
 			time.Sleep(1 * time.Second)
 		}
-
-		// nolint: errcheck
-		unix.Reboot(int(unix.LINUX_REBOOT_CMD_RESTART))
 	}
-
-	select {}
 }
 
 func main() {
+	defer reboot()
```

Comment:
Mind adding s comment above this around why it's being deferred and why it should be first. I'm expecting to preserve the defer execution stack we'd need to make sure `reboot()` is aways the last func executed?

Reply:
yes, exactly, deferred function are called in the reverse order, so `recovery()` handles any panic in main goroutine, and `reboot()` is executed anyways - after panic, or when `main()` function terminates (on clean shutdown/reboot)

I will add comment!

Reply:
I've added comments there, I hope they're correct